backend/src/test/java/com/nikogrid/backend/controllers/AuthenticationControllerTest.java

```diff
@@ -0,0 +1,178 @@
+package com.nikogrid.backend.controllers;
+
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nikogrid.backend.ErrorHandler;
+import com.nikogrid.backend.entities.RoleName;
+import com.nikogrid.backend.dto.AuthenticationDTO;
+import com.nikogrid.backend.dto.RecoveryJwtTokenDTO;
+import com.nikogrid.backend.dto.RegisterDTO;
+import com.nikogrid.backend.exceptions.DuplicateUserException;
+
+import com.nikogrid.backend.repositories.UserRepository;
+import com.nikogrid.backend.services.AuthenticationService;
+import com.nikogrid.backend.services.JwtTokenService;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.test.context.bean.override.mockito.MockitoBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.MediaType;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(
+        controllers = AuthenticationController.class,
+        excludeAutoConfiguration = {
+                    org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration.class
+        }
+)
+@Import(ErrorHandler.class)
+@DisplayName("AuthenticationController Tests")
+class AuthenticationControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockitoBean
+    private AuthenticationService authenticationService;
+
+    @MockitoBean
+    private AuthenticationManager authenticationManager;
+
+    @MockitoBean
+    private UserRepository userRepository;
+
+    @MockitoBean
+    private JwtTokenService jwtTokenService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    private static final String LOGIN_ENDPOINT = "/api/v1/auth/login";
+    private static final String REGISTER_ENDPOINT = "/api/v1/auth/register";
+
+    @Test
+    @DisplayName("Should authenticate user successfully")
```

Comment:
Add `@Requirement` annotation that links to the relevant user story in jira. Do this for all the tests.